concordium-node/src/grpc2.rs

```diff
@@ -2615,3 +2646,115 @@ where
         })
     }
 }
+
+/// Connection stream with an attached semaphore and gauge.
+pub struct ConnStreamWithTicket {
+    stream:                 TcpIncoming,
+    /// The semaphore is used to keep track of the number of connections we
+    /// currently maintain. For each new connection we give out a ticket.
+    semaphore:              tokio_util::sync::PollSemaphore,
+    /// A gauge to record the number of connected client. This is incremented
+    /// when we accept a new connection and decremented on the [`drop`] of
+    /// the connection.
+    grpc_connected_clients: GenericGauge<AtomicU64>,
```

Comment:
```suggestion
    grpc_connected_clients: IntGauge,
```

Reply:
https://docs.rs/prometheus/latest/prometheus/type.IntGauge.html

Not the same.